Makefile

```diff
@@ -48,6 +50,39 @@ else
 	docker build -t konstraint:latest -t konstraint:$(version) --build-arg KONSTRAINT_VER=$(version) .
 endif
 
+.PHONY: dockerx-build
+dockerx-build: ## Builds the docker image. Can optionally pass in a version.
+ifeq ($(version),)
+	docker buildx build \
+		--platform "$(PLATFORMS)" \
+		-t konstraint:latest \
+		.
+else
+	docker buildx build \
+		--platform "$(PLATFORMS)" \
+		-t konstraint:latest \
+		-t "konstraint:$(version)" \
+		--build-arg "KONSTRAINT_VER=$(version)" .
+endif
+
+.PHONY: dockerx-build-push
+dockerx-build-push: ## Builds and pushes the docker image. Can optionally pass in a version.
```

Comment:
Can this be updated to test the version and only push if a version is set? We don't want to push to GHCR except with a version set.

Reply:
Sure. My idea was to explicitly state in the make-command whether to push or not, since that could be any registry. But I'll remove dockerx-build-push and change dockerx-build to always push if there is a version if that works better for you.